src/era_validator.rs

```diff
@@ -61,57 +56,63 @@ pub fn era_validate(
-            epoch.to_string(),
-            master_accumulator.historical_epochs[epoch].0,
-        ) {
-            Ok(true) => {
-                log::info!("Skipping, epoch already synced: {}", epoch);
-                continue;
-            }
-            Ok(false) => {
-                log::info!("syncing new epoch: {}", epoch);
-            }
-            Err(e) => {
-                return {
-                    log::error!("error: {}", e);
-                    Err(EraValidateError::EpochAccumulatorError)
+        if use_lock {
+            match check_sync_state(
+                Path::new("./lockfile.json"),
+                epoch.to_string(),
+                master_accumulator.historical_epochs[epoch].0,
+            ) {
+                Ok(true) => {
+                    log::info!("Skipping, epoch already synced: {}", epoch);
+                    continue;
+                }
+                Ok(false) => {
+                    log::info!("syncing new epoch: {}", epoch);
+                }
+                Err(e) => {
+                    return {
+                        log::error!("error: {}", e);
+                        Err(EraValidateError::EpochAccumulatorError)
+                    }
                 }
             }
         }
-
-        let epoch_blocks: Vec<Block> = blocks.drain(0..MAX_EPOCH_SIZE).collect();
-        let root = process_blocks(epoch_blocks, epoch, &master_accumulator)?;
+        let epoch_headers: Vec<ExtHeaderRecord> = headers.drain(0..MAX_EPOCH_SIZE).collect();
+        let root = process_headers(epoch_headers, epoch, &master_accumulator)?;
         validated_epochs.push(epoch);
         // stores the validated epoch into lockfile to avoid validating again and keeping a concise state
-        match store_last_state(Path::new("./lockfile.json"), LockEntry::new(&epoch, root)) {
-            Ok(_) => {}
-            Err(e) => {
-                log::error!("error: {}", e);
-                return Err(EraValidateError::EpochAccumulatorError);
+        if use_lock {
+            match store_last_state(Path::new("./lockfile.json"), LockEntry::new(&epoch, root)) {
+                Ok(_) => {}
+                Err(e) => {
+                    log::error!("error: {}", e);
+                    return Err(EraValidateError::EpochAccumulatorError);
+                }
             }
         }
     }
 
     Ok(validated_epochs)
 }
 
-/// takes 8192 blocks and checks if they consist in a valid epoch
-fn process_blocks(
-    mut blocks: Vec<sf::ethereum::r#type::v2::Block>,
+/// takes 8192 block headers and checks if they consist in a valid epoch
+fn process_headers(
+    mut headers: Vec<ExtHeaderRecord>,
     epoch: usize,
     master_accumulator: &MasterAccumulator,
 ) -> Result<[u8; 32], EraValidateError> {
-    if blocks.len() != MAX_EPOCH_SIZE {
+    if headers.len() != MAX_EPOCH_SIZE {
         Err(EraValidateError::InvalidEpochLength)?;
     }
 
     if epoch > FINAL_EPOCH {
```

Comment:
I was confused for a minute about what happens with the epoch immediately before the merge. A clarifying comment would be helpful.